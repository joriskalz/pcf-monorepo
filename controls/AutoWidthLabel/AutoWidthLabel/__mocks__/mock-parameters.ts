/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
