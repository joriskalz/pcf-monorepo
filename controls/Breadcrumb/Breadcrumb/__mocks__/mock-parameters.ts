/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
