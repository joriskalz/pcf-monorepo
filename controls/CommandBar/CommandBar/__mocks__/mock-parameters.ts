/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
