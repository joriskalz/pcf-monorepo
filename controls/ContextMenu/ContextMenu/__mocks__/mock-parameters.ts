/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
