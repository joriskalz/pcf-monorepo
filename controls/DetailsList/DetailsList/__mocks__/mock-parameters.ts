/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
