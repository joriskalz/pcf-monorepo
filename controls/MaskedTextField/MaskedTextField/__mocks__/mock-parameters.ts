/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Value: new MockStringProperty(),
        Prefix: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        ErrorMessage: new MockStringProperty(),
        Mask: new MockStringProperty(),
        MaskFormat: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
    };
}
