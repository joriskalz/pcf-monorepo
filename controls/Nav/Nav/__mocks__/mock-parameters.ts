/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        CollapseByDefault: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
