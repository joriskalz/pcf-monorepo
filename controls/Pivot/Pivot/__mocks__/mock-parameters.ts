/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        RenderSize: new MockEnumProperty(),
        RenderType: new MockEnumProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
