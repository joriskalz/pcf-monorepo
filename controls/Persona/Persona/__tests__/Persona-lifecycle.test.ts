import { Persona } from '..';
import { IInputs } from '../generated/ManifestTypes';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { getMockParameters } from '../__mocks__/mock-parameters';
import * as renderer from 'react-test-renderer';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

jest.useFakeTimers();

describe('Persona', () => {
    beforeEach(() => jest.clearAllMocks());
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
    });

    it('renders', () => {
        const { component, context } = createComponent();
        component.init(context);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('theme', async () => {
        const { component, context } = createComponent();
        context.parameters.Theme.raw = JSON.stringify({
            palette: {
                themePrimary: '#0078d4',
            },
        });

        component.init(context);
        const personaComponent = renderer.create(component.updateView(context));
        expect(personaComponent.toJSON()).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new Persona();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.PersonaSize.raw = '18';
    context.parameters.ImageUrl.raw =
        'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png';
    context.parameters.Text.raw = 'Annie Lindqvist';
    context.parameters.ImageInitials.raw = 'AL';
    context.parameters.SecondaryText.raw = 'Software Engineer';
    context.parameters.TertiaryText.raw = 'In a meeting';
    context.parameters.OptionalText.raw = 'Available at 4:00pm';
    context.parameters.ImageAlt.raw = 'Annie Lindqvist, status is blocked';
    context.parameters.Presence.raw = '3';
    context.parameters.PersonaInitialsColor.raw = 'blue';
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, container, state };
}
