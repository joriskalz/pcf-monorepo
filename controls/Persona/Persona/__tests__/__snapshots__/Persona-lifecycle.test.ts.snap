// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persona renders 1`] = `
<Memo(CanvasPersona)
  ariaLabel=""
  imageAlt="Annie Lindqvist, status is blocked"
  imageInitials="AL"
  imageUrl="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png"
  initialsColor={1}
  optionalText="Available at 4:00pm"
  presence={3}
  secondaryText="Software Engineer"
  size={18}
  tertiaryText="In a meeting"
  text="Annie Lindqvist"
  themeJSON=""
/>
`;

exports[`Persona theme 1`] = `
<div
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    className=
        ms-Persona
        ms-Persona--size120
        ms-Persona--away
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          align-items: center;
          box-shadow: none;
          box-sizing: border-box;
          color: #323130;
          display: flex;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          height: 120px;
          margin-bottom: 0px;
          margin-left: 0px;
          margin-right: 0px;
          margin-top: 0px;
          min-width: 120px;
          padding-bottom: 0px;
          padding-left: 0px;
          padding-right: 0px;
          padding-top: 0px;
          position: relative;
        }
        & .contextualHost {
          display: none;
        }
  >
    <div
      className=
          ms-Persona-coin
          ms-Persona--size120
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
          }
      role="presentation"
    >
      <div
        className=
            ms-Persona-imageArea
            {
              flex: 0 0 auto;
              height: 120px;
              position: relative;
              text-align: center;
              width: 120px;
            }
        role="presentation"
      >
        <div
          className=
              ms-Image
              ms-Persona-image
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                border-radius: 50%;
                border: 0px;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                height: 120px;
                left: 0px;
                margin-right: 10px;
                overflow: hidden;
                perspective: 1px;
                position: absolute;
                top: 0px;
                width: 120px;
              }
          style={
            Object {
              "height": 120,
              "width": 120,
            }
          }
        >
          <img
            alt="Annie Lindqvist, status is blocked"
            className=
                ms-Image-image
                ms-Image-image--cover
                ms-Image-image--portrait
                is-notLoaded
                is-fadeIn
                {
                  display: block;
                  height: 100%;
                  object-fit: cover;
                  opacity: 0;
                  width: 100%;
                }
            onError={[Function]}
            onLoad={[Function]}
            src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/persona-female.png"
          />
        </div>
        <div
          className=
              ms-Persona-presence
              {
                -ms-high-contrast-adjust: none;
                background-clip: border-box;
                background-color: #FFAA44;
                border-radius: 50%;
                border: 2px solid #ffffff;
                bottom: -2px;
                box-sizing: content-box;
                forced-color-adjust: none;
                height: 32px;
                position: absolute;
                right: -2px;
                text-align: center;
                top: auto;
                width: 32px;
              }
              @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                background-color: WindowText;
                border-color: Window;
              }
          role="presentation"
        >
          <i
            aria-hidden={true}
            className=
                ms-Persona-presenceIcon
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  color: #ffffff;
                  display: inline-block;
                  font-family: "FabricMDL2Icons";
                  font-size: 14px;
                  font-style: normal;
                  font-weight: normal;
                  left: 1px;
                  line-height: 32px;
                  position: relative;
                  speak: none;
                  vertical-align: top;
                }
                @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
                  color: Window;
                }
            data-icon-name="SkypeClock"
          >
            ÓæÅ
          </i>
        </div>
      </div>
    </div>
    <div
      className=
          ms-Persona-details
          {
            display: flex;
            flex-direction: column;
            justify-content: space-around;
            min-width: 0px;
            padding-bottom: 0;
            padding-left: 16px;
            padding-right: 24px;
            padding-top: 0;
            text-align: left;
            width: 100%;
          }
    >
      <div
        className=
            ms-Persona-primaryText
            {
              color: #323130;
              font-size: 20px;
              font-weight: 400;
              overflow: hidden;
              text-overflow: ellipsis;
              white-space: nowrap;
            }
            &:hover {
              color: #201f1e;
            }
        dir="auto"
      >
        <div
          className=
              ms-TooltipHost
              {
                display: inline;
              }
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          Annie Lindqvist
        </div>
      </div>
      <div
        className=
            ms-Persona-secondaryText
            {
              color: #605e5c;
              font-size: 14px;
              font-weight: 400;
              overflow: hidden;
              text-overflow: ellipsis;
              white-space: nowrap;
            }
        dir="auto"
      >
        <div
          className=
              ms-TooltipHost
              {
                display: inline;
              }
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          Software Engineer
        </div>
      </div>
      <div
        className=
            ms-Persona-tertiaryText
            {
              color: #605e5c;
              display: block;
              font-size: 14px;
              font-weight: 400;
              overflow: hidden;
              text-overflow: ellipsis;
              white-space: nowrap;
            }
        dir="auto"
      >
        <div
          className=
              ms-TooltipHost
              {
                display: inline;
              }
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          In a meeting
        </div>
      </div>
      <div
        className=
            ms-Persona-optionalText
            {
              color: #605e5c;
              display: block;
              font-size: 14px;
              font-weight: 400;
              overflow: hidden;
              text-overflow: ellipsis;
              white-space: nowrap;
            }
        dir="auto"
      >
        <div
          className=
              ms-TooltipHost
              {
                display: inline;
              }
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          Available at 4:00pm
        </div>
      </div>
    </div>
  </div>
</div>
`;
